m4/slepc.m4

```diff
@@ -16,26 +16,47 @@ AC_DEFUN([CONFIGURE_SLEPC],
 
   AS_IF([test "$enableslepc" !=  no],
         [
-          dnl Test to see if SLEPC_DIR set by user.  If not set, then
-          dnl try to autodetect in a default directory
+          dnl If SLEPC_DIR is not set, we should try PETSC_DIR
+          dnl since SLPEc could live in the same place as PETSc
           AS_IF([test "x$SLEPC_DIR" = x],
-                [export SLEPC_DIR=/usr/lib/slepc])
+                [export SLEPC_DIR=$PETSC_DIR])
 
           dnl Test to see if SLEPC_DIR and petscversion were set by user or
           dnl autodetection.  If not set, then disable slepc, print a message.
           AS_IF([test "x$SLEPC_DIR" = "x" || test "x$petscversion" = "x"],
-                [
-                  enableslepc=no
-                  AC_MSG_RESULT(<<< SLEPc disabled.  Please set your "\$SLEPC_DIR" environment variable correctly to enable SLEPc. >>>)
-                ],
-                [
-                  AC_CHECK_HEADER([$SLEPC_DIR/include/slepcversion.h],
-                                  [SLEPC_INCLUDE="-I$SLEPC_DIR/include"],
+                 [
+                   dnl Test to see if SLEPC_DIR set by user.  If not set, then
+                   dnl try to autodetect in a default directory
+                   AC_CHECK_HEADER([/usr/lib/slepc/slepcversion.h],
```

Comment:
Looking in `/usr/lib/slepc` as a default is what we did before, but is that really what we want to do in general? 

It seems that it runs the risk of picking up some system installation of SLEPc when the user didn't want SLEPc enabled at all. If this ever happened it would probably be extremely rare, but also very confusing to the user.

Reply:
> it would probably be extremely rare,

/usr/lib/slepc is the default install directory in Debian and Ubuntu.

> but also very confusing to the user.

If they installed slepc, I think using it would be natural and failing to do so would be confusing, no?

Reply:
> If they installed slepc, I think using it would be natural and failing to do so would be confusing, no?

Installed SLEPc but *didn't* set `$SLEPC_DIR`? IMO we can't assume they want SLEPc just because it's on their system, sometimes you don't have 100% control over what's installed on your system...

Reply:
I think if you do not have SLEPC_DIR, and do not have PETSC_DIR, and do not enable SLEPc, then we should not do anything.

I want to let libMesh take a try to check if PETSc is complied with SLEPc. Then SLEPc will be picked up in the same place as PETSc.  I assume if users pass ``--download-slepc" in PETSc, then users want to use SLEPc. Note that ` --download-slepc` does not set up SLEPC_DIR.

Reply:
I'll consider myself outvoted.  I still think asking users who installed-but-don't-want SLEPc to use "--disable-slepc" is more user-friendly than asking users who installed-and-want SLEPc to manually specify paths.  Should we make the same change to PETSc autodetection for consistency?  Or for **real** consistency, I notice that we also automatically activate CPPUnit, curl, GLPK, Tecplot, Trilinos, VTK, and XDR based on what we find in /usr...  ;-)